usr/src/psm/stand/boot/aarch64/rpi4/machdep.c

```diff
@@ -93,6 +93,25 @@ setup_aux(void)
 {
 }
 
+/*
+	Remove -D and the path from the boot args, look for the next "-" char
+*/
+void
+fix_boot_args(const char *str)
+{
+        char *s2;
+        char *s3;
+        int i;
+        s2 = strstr(str, "-D");
+        if(s2 != NULL){
+                s3 = s2+2;
+                for(i=0; s3[i]!=0 && s3[i]!='-'; i++);
+                s3 = s3 + i;
+                memmove(s2, s3, strlen(s3)+1);
```

Comment:
Can you cstyle at least the bits you're adding?  I know the whole file is in a state.
I'm a bit unsure about `str` being const, given we're changing it, even though we're changing it in a way compatible with the `const`, it seems like it might confuse people.